primitives/src/channel_validator.rs

```diff
@@ -1,11 +1,16 @@
 use crate::channel::{Channel, ChannelError, SpecValidator, SpecValidators};
 use crate::config::Config;
+use crate::ValidatorId;
 use chrono::Utc;
+use std::convert::TryFrom;
 
 pub trait ChannelValidator {
     fn is_channel_valid(config: &Config, channel: &Channel) -> Result<(), ChannelError> {
         let identity = &config.clone().identity.unwrap_or_else(|| "".to_string());
-        let adapter_channel_validator = match channel.spec.validators.find(identity) {
+        let validator_identity = ValidatorId::try_from(&identity[..]).map_err(|_| {
```

Comment:
I think `&identity` is enough here or is there another reason for this?

Reply:
identity is a `String` appending it with `[...]` is a shortcut for converting it into a `str` slice

Reply:
TryFrom is implemented for `&str`

Reply:
Yes, but you have a Deref for String, which means you can pass `&string` and it will convert it to `&str`

Reply:
No, you can't, it's not possible. It requires implementing TryFrom for &String

Reply:
You can read more here 
https://stackoverflow.com/questions/48244892/should-rust-implementations-of-from-tryfrom-target-references-or-values

Reply:
ok, then why not just use `.as_str()`, and btw you don't need the `&` on the line above, you have an owned `String` there